doc/user-guide/How-to-build.md

```diff
@@ -1,71 +1,116 @@
-*   Zlib 1.2.3 or higher for Stream Compression support (XEP-0138). Optional.
+    * CentOS: `unixODBC-devel`
+    * Ubuntu: `unixodbc-dev`
+    * Mac (Homebrew): `unixodbc`
+*   Zlib 1.2.3 or higher
+    * CentOS: `zlib-devel`
+    * Ubuntu: `zlib1g-dev`
+    * Mac: built-in
+
+## Preparing macOS environment
 
-## Compiling on macOS
+### Step 1
 
-MongooseIM is supported on Mac OS X / macOS 10.6.8 and later.
-Before you can compile and run MongooseIM, you will need to install the following packages:
+Install [Homebrew](http://brew.sh) to manage packages on your Mac.
+You may use a different package manager but you'll need to figure out the package names and file paths on your own.
 
-* C and C++ compiler
-* Git
-* Erlang/OTP 18.3 or higher
-* OpenSSL
-* ODBC
+### Step 2
 
-We recommend using [Homebrew](http://brew.sh) to manage packages on your Mac.
-With Homebrew installed, getting all dependencies is a matter of running the following commands:
+Install Xcode Command Line Tools.
 
 ```bash
 $ xcode-select --install # install compilation tools
-$ brew install git erlang openssl unixodbc
 ```
 
-If you are planning to use ODBC to connect to MSSQL:
+### Step 3
+
+Install dependencies with Brew.
+
+```
+$ brew install erlang openssl unixodbc
+```
+
+### Step 4
+
+Add OpenSSL paths to the compiler and linker environment variables:
 
 ```bash
-$ brew install freetds --with-unixodbc
+$ export LDFLAGS="-L/usr/local/Cellar/openssl/[installed version]/lib/ -undefined dynamic_lookup $LDFLAGS"
+$ export CFLAGS="-I/usr/local/Cellar/openssl/[installed version]/include/ $CFLAGS"
 ```
 
-Additionally you'll need to export some environment variables to point the compiler to OpenSSL
-installed with Homebrew:
+Please remember to replace `[installed version]` with the one that is present in your file system.
+1.0.2s was the most recent one when this guide was written.
+
+Now, please proceed to the "Building" section.
+
+## Preparing CentOS environment
+
+Please install required dependencies:
 
 ```bash
-$ export LDFLAGS="-L/usr/local/opt/openssl/lib -undefined dynamic_lookup $LDFLAGS"
-$ export CXXFLAGS="-I/usr/local/opt/openssl/include $CXXFLAGS"
-$ export CFLAGS="-I/usr/local/opt/openssl/include $CFLAGS"
+$ sudo yum install git make zlib-devel unixODBC-devel gcc gcc-c++ erlang
 ```
 
-After you've done this, follow the steps from the [Compiling on *nix systems](#compiling-on-42nix-systems) section.
+Now, please proceed to the "Building" section.
+
+## Preparing Ubuntu environment
 
-## Compiling on \*nix systems
+Please install required dependencies:
 
-To compile MongooseIM, go to the main repo directory `$REPO` and execute the command (`$` stands for the shell prompt):
+```bash
+$ sudo apt install git make zlib1g-dev unixodbc-dev gcc g++ erlang
+```
 
+Now, please proceed to the "Building" section.
 
-        $ ./rebar3 compile
+## Building
 
-To generate a full MongooseIM release (i.e. an executable with mysql, pgsql or other dependencies):
```

Comment:
To generate a full MongooseIM release (i.e. one that is executable with mysql, pgsql or other dependencies):